docs/openshift/operator-marketplace.md

```diff
@@ -0,0 +1,69 @@
+# Operator Marketplace
+
+On OpenShift clusters, the [marketplace-operator](https://github.com/operator-framework/operator-marketplace) makes off-cluster Quay Appregistry type catalogs available to OLM. It is installed by default with OpenShift.
+
+## Pulling Catalogs From Appregistry
+
+The marketplace-operator can be told to pull a catalog from Appregistry by creating an `OperatorSource`:
+
+```yaml
+apiVersion: operators.coreos.com/v1
+kind: OperatorSource
+metadata:
+  name: my-operators
+  namespace: openshift-marketplace
+spec:
+  type: appregistry
+  endpoint: https://quay.io/cnr
+  registryNamespace: my-operators
+  displayName: "My Operators"
+  publisher: "Me, Myself, and I"
+```
+
+- `type` is the registry type (`appregistry` being the only supported type)
+- `endpoint` is the Appregistry endpoint
+- `registryNamespace` is the name of your Appregistry namespace
+- `displayName` and `publisher` are optional and only needed for UI purposes
+
+The creation of an `OperatorSource` results in the creation of an OLM `CatalogSource` in the same namespace the marketplace-operator is running in. This `CatalogSource` will be populated with operators from the `OperatorSource` ready to be managed by OLM.
+
+In versions of OpenShift > 4.2, all `CatalogSources` generated by the marketplace-operator in the `openshift-marketplace` namespace are available for use in all namespaces.
+
+## Configuration
+
+The marketplace-operator is installed by default into the `openshift-marketplace` namespace along with three Red Hat provided `OperatorSources`; `redhat-operators`, `certified-operators`, and `community-operators`, which represent operators published by redhat, ISVs (Independent Software Vendors), and upstream community members respectively.
+
+In versions of OpenShift > 4.2, The default behavior of the marketplace-operator ensures these `OperatorSources` always exist on a cluster, even if they are manually deleted.
+
+You can manually configure which `OperatorSources` the `marketplace-operator` recreates by creating an `OperatorHub` resource:
+
+```yaml
+apiVersion: config.openshift.io/v1
+kind: OperatorHub
+metadata:
+  name: cluster
+spec:
+  disableAllDefaultSources: true
+  sources: [
+    {
+      name: "community-operators",
+      disabled: false,
+    }
+  ]
+```
+
+In OpenShift 4.1, recreation of the default `OperatorSources` is controlled by [CVO (cluster-version-operator)](https://github.com/openshift/cluster-version-operator/tree/release-4.1) and can be disabled by [setting them as unmanaged resources](https://github.com/openshift/cluster-version-operator/blob/release-4.1/docs/dev/clusterversion.md#setting-objects-unmanaged):
+
+```sh
+$ cat <<EOF >version-patch.yaml
+> - op: add
+>   path: /spec/overrides/-
+>   value:
+>     kind: OperatorSource
+>     name: community-operators
+>     namespace: openshift-marketplace
+>     unmanaged: true
+> EOF
+$ kubectl patch clusterversion version --type json -p "$(cat version-patch.yaml)"
```

Comment:
There's one more step to this. After you mark it as unmanaged you still need to:

```sh
oc delete opsrc community-operators -n=openshift-marketplace
```

It may seem obvious, but I think we want to explicitly call out that, unlike the config object which will cause the marketplace operator to delete `community-operators`, the cvo will just stop recreating it automatically.